.vale/styles/OpenShiftAsciiDoc/ModuleContainsContentType.yml

```
@@ -1,8 +1,29 @@
 ---
-extends: occurrence
-scope: raw
+extends: script
+message: "Module is missing the \"_content-type\" variable."
 level: error
 link: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#module-file-metadata
-message: "Module is missing the \"_content-type\" variable."
-min: 1
-token: '(?<!\/\/):_content-type:(\s)?(CONCEPT|PROCEDURE|REFERENCE|ASSEMBLY|SNIPPET)'
+scope: raw
+script: |
+  text := import("text")
+  matches := []
+
+  //trim extra whitespace
+  scope = text.trim_space(scope)
+  //add a newline, it might be missing
+  scope += "\n"
+
+  context_regex := ":_content-type:(\\s)?(CONCEPT|SNIPPET|ASSEMBLY|PROCEDURE|REFERENCE)"
```

Pretty sure the variable needs precisely one space after the variable name or AsciiDoctor complains

---

```suggestion
  context_regex := "^:_(mod-docs-)?content-type:\\s+(CONCEPT|SNIPPET|ASSEMBLY|PROCEDURE|REFERENCE)"
```

The `modular-docs` project [recently changed](https://github.com/redhat-documentation/modular-docs/pull/209) the names of the attributes identifying the module type.

---

Thanks @jhradilek - I don't think OpenShift have yet implemented this change, but when it does, I will need to update this rule

---

> Pretty sure the variable needs precisely one space after the variable name or AsciiDoctor complains

Update. Does this work?

---

IMO - if there is a failing on the VS Code plugin side, I would prefer to see that corrected rather than working around it with this update.

---

Created https://github.com/ChrisChinchilla/vale-vscode/issues/14